src/android/Peripheral.java

```diff
@@ -80,10 +79,19 @@ public void connect(CallbackContext callbackContext, Activity activity) {
     }
 
     public void disconnect() {
-        connectCallback = null;
         connected = false;
         connecting = false;
 
+		if (connectCallback != null) {
```

Comment:
The connect error callback shouldn't be called here since the user requested the peripheral disconnection.

Reply:
I think it should, since the expectation when you call a function with success and failure callbacks is that one or the other will eventually fire. This shows up particularly when used with Ionic Native, since that wraps Cordova callbacks in Promises or Observables, and then you have a Promise that never resolves/Observable that never terminates.